### ISSUE TITLE
How should we return relevant status codes. 

### ISSUE THREAD

**Issue by Ryan-Gordon** [exegesis-js/exegesis#20]
In the 2 samples using Exegesis. One endpoint returns a string with a 200 status code. 

What would be the recommended way to return a desired status code. 

Taking an example of searching a database for a value that isn't there, one might return a 404 or 400 status code to the user depending on whether they are following a REST spec.
How can we the dev, catch any potential errors and then send a 404/400/XXX status code.

I found one method which would be to take an instance of ExegesisContext as a parameter to the controller being used and calling `context.res.setStatus(404)` before returning any values but would this be the preferred way of doing things?

**Comment by jwalton** [comment]
Yeah, exactly.  Or `context.res.status()`, which is chainable.  Something like:

```js
export async function getUser(context) {
    const user = await db.getUser({_id: context.params.query.id});
    if(!user) {
        // `end()` because we don't want to send back a body.
        context.res
            .status(404)
            .end();
    } else {
        context.res
            .status(200)
            .setHeader('content-type', 'application/json');
            .setBody(user);
    }
}
```

**Comment by Ryan-Gordon** [comment]
Great to have 2 options for it ! 
I can see quite a few handy uses for `context.res.status()`

Closing this now as that fully answers the query, thank you!
